src/dotplot.jl

```diff
@@ -0,0 +1,91 @@
+
+# ---------------------------------------------------------------------------
+# Dot Plot
+
+@recipe function f(::Type{Val{:dotplot}}, x, y, z; widthwindow=0.1)
+    # if only y is provided, then x will be UnitRange 1:length(y)
+    if typeof(x) <: AbstractRange
+        if step(x) == first(x) == 1
+            x = plotattributes[:series_plotindex]
+        else
+            x = [getindex(x, plotattributes[:series_plotindex])]
+        end
+    end
+    glabels = sort(collect(unique(x)))
+    warning = false
+    points_x, points_y = zeros(0), zeros(0)
+    bw = plotattributes[:bar_width]
+    bw == nothing && (bw = 0.8)
+    0.0 ≤ widthwindow ≤ 1.0 || throw(ArgumentError("$(:widthwindow) must be in the range [0,1]"))
+    for (i,glabel) in enumerate(glabels)
+        # filter y
+        values = y[filter(i -> _cycle(x,i) == glabel, 1:length(y))]
+
+        # compute quantiles
+        q1,q2,q3,q4,q5 = quantile(values, Base.range(0,stop=1,length=5))
+
+        # make the shape
+        center = Plots.discrete_value!(plotattributes[:subplot][:xaxis], glabel)[1]
+        hw = 0.5_cycle(bw, i) # Box width
+
+        nearby = widthwindow * abs(q5 - q1)
+        countnear = [count((values .< (x + nearby)) .& (values .> (x - nearby))) for x ∈ values]
+
+        pw = hw / maximum(countnear)
+        offsets = [(rand() * 2 - 1) * pw * x for x ∈ countnear]
```

Comment:
This means that plots with the same data will be different each time, right?
Do you think there is also a unique method of distributing the points vertically?